plugin/adsk/scripts/AETemplateHelpers.py

```diff
@@ -11,6 +12,46 @@ def debugMessage(msg):
     if DEBUG:
         print(msg)
 
+def GetAllRootPrimNamesNaturalOrder(proxyShape):
+    # Custom comparator
+    def naturalOrderComparator(a, b):
+        lenA, lenB = len(a), len(b)
+        i, j = 0, 0
+        while((i < lenA) and (j < lenB)):
+            if(a[i].isdigit() and b[j].isdigit()):
+                # extract the number part and put into a separate string
+                numA, numB = str(a[i]), str(b[j])
+                while((i + 1 < lenA) and a[i+1].isdigit()):
+                    i += 1
+                    numA += a[i]
+                while((j + 1 < lenB) and b[j+1].isdigit()):
+                    j += 1
+                    numB += b[j]
+
+                if(numA == numB):
+                    continue
+                return int(numA) - int(numB)
```

Comment:
If you keep your version, you wil lhave to fix that you are returning and int here but returning boolean (True, False) saying if a is greater or not at line 38, but return (True, False) saying if they are equal or not at line 39!

Reply:
(I just tested your version, and it goes into an infinite loop if both strings are equal but end with digits. I guess root prim names never are equal, so that is why it happens to work).